index.js

```diff
@@ -156,4 +148,28 @@ function createExtensionMap(exts) {
   return exts;
 }
 
+function isEmpty(object) {
+  return !Object.keys(object).length;
+}
+
+function isString(value) {
+  if (typeof value === 'string') {
+    return true;
+  }
+
+  if (Object.prototype.toString.call(value) === '[object String]') {
+    return true;
+  }
+
+  return false;
+}
+
+function defaultsNotNullish(mainObj, defaultObj) {
+  return filterValues(defaults(mainObj, defaultObj), isNotNullish);
```

Comment:
if there cases where the user might want to set a value to `null` or `undefined` intentionally, to overwrite some other value, then this will prevent that from working.